creator-node/src/services/stateMachineManager/stateReconciliation/StateReconciliationQueue.js

```diff
@@ -1,5 +1,273 @@
+
+  makeQueue(redisHost, redisPort) {
+    // Settings config from https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#advanced-settings
+    return new BullQueue(STATE_RECONCILIATION_QUEUE_NAME, {
+      redis: {
+        host: redisHost,
+        port: redisPort
+      },
+      defaultJobOptions: {
+        removeOnComplete: QUEUE_HISTORY,
+        removeOnFail: QUEUE_HISTORY
+      },
+      settings: {
+        // Should be sufficiently larger than expected job runtime
+        lockDuration: STATE_RECONCILIATION_QUEUE_MAX_JOB_RUNTIME_MS,
+        // We never want to re-process stalled jobs
+        maxStalledCount: 0
+      }
+    })
+  }
+
+  /**
+   * Registers event handlers for logging and job success/failure.
+   * @param {Object} params.queue the queue to register events for
+   * @param {Function<queue, successfulJob, jobResult>} params.jobSuccessCallback the function to call when a job succeeds
+   * @param {Function<queue, failedJob>} params.jobFailureCallback the function to call when a job fails
+   * @param {Function<job>} params.processManualSync the function to call when processing a manual sync job from the queue
+   * @param {Function<job>} params.processRecurringSync the function to call when processing a recurring sync job from the queue
+   * @param {Function<job>} params.processIssueSyncRequests the function to call when processing an issue-sync-requests job from the queue
+   * @param {Function<job>} params.processUpdateReplicaSet the function to call when processing an update-replica-set job from the queue
+   */
+  registerQueueEventHandlersAndJobProcessor({
+    queue,
+    processManualSync,
+    processRecurringSync,
+    processIssueSyncRequests,
+    processUpdateReplicaSets
+  }) {
+    // Add handlers for logging
+    queue.on('global:waiting', (jobId) => {
+      this.log(`Queue Job Waiting - ID ${jobId}`)
+    })
+    queue.on('global:active', (jobId, jobPromise) => {
+      this.log(`Queue Job Active - ID ${jobId}`)
+    })
+    queue.on('global:lock-extension-failed', (jobId, err) => {
+      this.logError(
+        `Queue Job Lock Extension Failed - ID ${jobId} - Error ${err}`
+      )
+    })
+    queue.on('global:stalled', (jobId) => {
+      this.logError(`stateMachineQueue Job Stalled - ID ${jobId}`)
+    })
+    queue.on('global:error', (error) => {
+      this.logError(`Queue Job Error - ${error}`)
+    })
+
+    // Add handlers for when a job fails to complete (or completes with an error) or successfully completes
+    queue.on('completed', (job, result) => {
+      this.log(
+        `Queue Job Completed - ID ${job?.id} - Result ${JSON.stringify(
+          result
+        )}. Queuing another job...`
+      )
+    })
+    queue.on('failed', (job, err) => {
+      this.logError(`Queue Job Failed - ID ${job?.id} - Error ${err}`)
+    })
+
+    // Register the logic that gets executed to process each new job from the queue
+    queue.process(
+      HANDLE_MANUAL_SYNC_REQUEST_QUEUE_NAME,
```

Comment:
nit naming - these aren't names of diff queues, but rather names of diff job processors on single queue